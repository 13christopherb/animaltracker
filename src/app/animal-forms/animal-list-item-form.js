import React from 'react';
import {Field} from 'redux-form'
import {Form, Col, Button} from 'react-bootstrap'
import {Select} from "../form-fields/index";

/**
 * Form for creating a new animal
 * @param handleSubmit
 * @param handleChange
 * @param onSubmit
 * @param props
 * @returns {*}
 * @constructor
 */
export const AnimalListItemForm = ({handleSubmit, handleChange, onSubmit, ...props}) => {
    return (
        <div>
            <Form onSubmit={handleSubmit(onSubmit)}>
                <Form.Row>
                    <Col md={2}>
                        <h6>{props.animal.name}</h6>
                        <Field
                            component={Select}
                            id="location"
                            name="location"
                            value={props.value}
                            onChange={props.handleChange}
                            options={[
                                'NRO',
                                'MBO',
                                'SLO'
                            ]}
                        />
                    </Col>
                    <Col md={6}>
                        <Button variant="primary" disabled={props.pristine || !props.valid} type="submit">
                            Submit
                        </Button>
                        <Button onClick={props.toggleMoving} variant="danger" type="button">
                            Cancel
                        </Button>
                    </Col>
                </Form.Row>
            </Form>
        </div>
    )
};