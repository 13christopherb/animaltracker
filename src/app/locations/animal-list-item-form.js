import React from 'react';
import {Field} from 'redux-form'
import {Input, Select} from "../form-fields/index";

/**
 * Form for creating a new animal
 * @param handleSubmit
 * @param handleChange
 * @param onSubmit
 * @param props
 * @returns {*}
 * @constructor
 */
export const AnimalListItemForm = ({handleSubmit, handleChange, onSubmit, ...props}) => {
    return (
        <div>
            <form onSubmit={handleSubmit(onSubmit)}>
                <div className="form-row">
                <div className="col col-md-2">
                    <h6>{props.animal.name}</h6>
                    <Field
                        component={Select}
                        id="location"
                        name="location"
                        value={props.value}
                        onChange={props.handleChange}
                        options={[
                            'NRO',
                            'MBO',
                            'SLO'
                        ]}
                    />
                </div>
                <div className="col col-md-6">
                    <button className="btn btn-primary" disabled={props.pristine || !props.valid} type="submit">
                        Submit
                    </button>
                    <button onClick={props.toggleAddAnimal} className="btn btn-danger" type="button">
                        Cancel
                    </button>
                </div>
                </div>
            </form>
        </div>
    )
};