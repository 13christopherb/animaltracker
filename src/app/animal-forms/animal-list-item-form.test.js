import React from 'react';
import configureMockStore from 'redux-mock-store';
import {mount} from 'enzyme';
import {AnimalListItemForm} from "./animal-list-item-form";
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import {reduxForm} from 'redux-form'

const mockStore = configureMockStore();


jest.mock('react-dom');

const onSubmit = jest.fn();
const Decorated = reduxForm({
    form: 'testForm', onSubmit: {onSubmit}
})(AnimalListItemForm);

describe('<AnimalListeItemForm />', () => {
    const defaultProps = {
        pristine: false,
        valid: true,
        animal: {
            name: 'Kirt',
            species: 'CSL',
            location: 'MBO',
            weight: 42
        }
    };
    test('AnimalForm renders correctly', () => {
        const store = mockStore();
        const tree = renderer.create(
            <Provider store={store}>
                <Decorated
                    {...defaultProps}
                />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot()
    });
    test('Disables button on form errors', () => {
        const store = mockStore();
        const props = {
            ...defaultProps,
            valid: false,
        };
        const wrapper = mount(<Provider store={store}>
            <Decorated
                {...props}
            />
        </Provider>);
        expect(wrapper.find('button[type="submit"]').props().disabled).toEqual(true);
    });
});