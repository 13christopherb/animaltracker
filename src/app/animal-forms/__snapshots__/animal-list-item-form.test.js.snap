// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimalListeItemForm /> AnimalForm renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-row"
    >
      <div
        className="col col-md-2"
      >
        <h6>
          Kirt
        </h6>
        <div
          className="form-group"
        >
          <label />
          <select
            className="custom-select "
            id="location"
            name="location"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option
              disabled={true}
              value=""
            />
            <option
              label="NRO"
              value="NRO"
            >
              NRO
            </option>
            <option
              label="MBO"
              value="MBO"
            >
              MBO
            </option>
            <option
              label="SLO"
              value="SLO"
            >
              SLO
            </option>
          </select>
        </div>
      </div>
      <div
        className="col col-md-6"
      >
        <button
          className="btn btn-primary"
          disabled={true}
          type="submit"
        >
          Submit
        </button>
        <button
          className="btn btn-danger"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
</div>
`;
